cupy/core/core.pyx

```diff
@@ -1742,6 +1742,45 @@ cpdef ndarray ascontiguousarray(ndarray a, dtype=None):
     elementwise_copy(a, newarray)
     return newarray
 
+
+cpdef ndarray asfortranarray(ndarray a, dtype=None):
+    cdef ndarray newarray
+    cdef int m, n
+
+    if dtype is None:
+        if a._f_contiguous:
+            return a
+        dtype = a.dtype
+    else:
+        dtype = numpy.dtype(dtype)
+        if a._f_contiguous and dtype == a.dtype:
+            return a
+
+    newarray = ndarray(a.shape, dtype, order='F')
+    if (a.flags.c_contiguous and
+            (a.dtype == numpy.float32 or a.dtype == numpy.float64) and
+            a.ndim == 2 and dtype is None):
```

Comment:
Always `dtype is not None` https://github.com/pfnet/chainer/pull/2176/files#diff-bebe923819e806ae22e9923de58e954cR1753